src/lib/core/say.c

```diff
@@ -906,7 +911,15 @@ say_format_json(struct log *log, char *buf, int len, int level,
 		const char *module, const char *filename, int line,
 		const char *error, const char *format, va_list ap)
 {
-	(void)log;
+	str_escape_t escape;
+	str_escape_inplace_t escape_inplace;
+	if (log->type != SAY_LOGGER_SYSLOG) {
+		escape = json_escape;
+		escape_inplace = json_escape_inplace;
+	} else {
+		escape = json_syslog_escape;
+		escape_inplace = json_syslog_escape_inplace;
+	}
```

Comment:
IMO having a separate function that escapes a message according to json+syslog is an overkill. Also, I don't like that this patch makes the log format callbacks depend on the log type even more. Why not simply escape the resulting string with `syslog_escape_inplace` in `format_log_entry`, where we add a syslog header? AFAICS escaping a string for syslog should be valid for any log format.